services/core/shared/core/compat/sdk_v5/blocks.js

```diff
@@ -310,12 +320,15 @@ const buildIndex = async (from, to) => {
 		/* eslint-disable no-await-in-loop */
 		const pseudoOffset = to - (MAX_BLOCKS_LIMIT_PP * (pageNum + 1));
 		const offset = pseudoOffset > from ? pseudoOffset : from - 1;
-		logger.info(`Attempting to cache blocks ${offset + 1}-${offset + MAX_BLOCKS_LIMIT_PP}`);
+		const batchFromHeight = offset + 1;
+		const batchToHeight = (offset + MAX_BLOCKS_LIMIT_PP) <= to
+			? (offset + MAX_BLOCKS_LIMIT_PP) : to;
+		logger.info(`Attempting to cache blocks ${batchFromHeight}-${batchToHeight}`);
 
 		let blocks;
 		do {
-			blocks = await getBlocksByHeightBetween(offset + 1, offset + MAX_BLOCKS_LIMIT_PP);
-		} while (!(blocks.length && blocks.every(block => !!block && !!block.height)));
+			blocks = await getBlocksByHeightBetween(batchFromHeight, batchToHeight);
+		} while (!(blocks.length && blocks.every(block => !!block && block.height >= 0)));
```

Comment:
Please consider updating `bottomHeightFromBatch` (line 338) to:
```
const bottomHeightFromBatch = sortedBlocks.length ? (sortedBlocks.shift()).height : topHeightFromBatch;
```
This handles an edge case scenario when the `sortedBlocks` array contains a single entry.

Reply:
Done!